googlepubsub/README.md

```diff
@@ -0,0 +1,170 @@
+          allowedPersistenceRegions:
+          - us-central1
+          - us-central2
+          - us-east1
+          - us-east4
+          - us-east5
+          - us-east7
+          - us-south1
+          - us-west1
+          - us-west2
+          - us-west3
+          - us-west4
+        schemaSettings:
+          encoding: binary
+          name: projects/your-project/schemas/message-proto
+# ...
+```
+
+<a name="message"></a>
+
+## Message Binding Object
+
+The `Message Binding Object` is used to describe the Google Cloud Pub/Sub specific
+[PubsubMessage](https://cloud.google.com/pubsub/docs/reference/rest/v1/PubsubMessage) details, alongside with pertintent
+parts of the Google Cloud Pub/Sub
+[Schema](https://cloud.google.com/pubsub/docs/reference/rest/v1/projects.schemas#Schema)
+Object, with AsyncAPI.
+
+Field Name | Type | Description
+---|---|---
+`bindingVersion`|String|The current version is `0.1.0`
+`attributes`|Object|Attributes for this message _(If this field is empty, the message must contain non-empty data. This can be used to filter messages on the subscription.)_
+`orderingKey`|String|If non-empty, identifies related messages for which publish order should be respected _(For more information, see [ordering messages](https://cloud.google.com/pubsub/docs/ordering).)_
+`schema`|[Schema Definition Object](#schema-definition-object)|Describes the schema used to validate the payload of this message
+
+<a name="schema-definition-object"></a>
+
+### Schema Definition Object
+
+The `Schema Definition Object` is used to describe the Google Cloud Pub/Sub
+[Schema]([Schema](https://cloud.google.com/pubsub/docs/reference/rest/v1/projects.schemas#Schema)) Object with AsyncAPI.
+While some of this information could be, or is, described using native AsyncAPI, for consistency it makes sense to
+provide this information here at all times, especially for cases where AsyncAPI does not natively support describing
+payloads using a supported Google Cloud Pub/Sub schema format like Protobuf.
+
+Field Name | Type | Description
+---|---|---
+`name`|String|The name of the schema
+`type`|String|The type of the schema
+
+<a name="message-binding-example"></a>
+
+### Example
+
+```yaml
+# ...
+components:
+  messages:
+    messageAvroAsJSON:
+      bindings:
+        googlepubsub:
+          schema:
+            name: projects/your-project/schemas/message-avro
+            type: avro
+      contentType: application/json
+      name: projects/your-project/schemas/message-avro
+      payload:
+        fields:
+        - name: message
+          type: string
+        name: Message
+        type: record
+      schemaFormat: application/vnd.apache.avro+yaml;version=1.9.0
+    messageProto:
+      bindings:
+        googlepubsub:
+          schema:
+            name: projects/your-project/schemas/message-proto
+            type: protobuf
+      contentType: application/octet-stream
+      name: projects/your-project/schemas/message-proto
```

Comment:
Would you please change the `name` to something that doesn't look like a Google Cloud object name? Message name must be "machine-friendly" but having the same value as `bindings.googlepubsub.schema.name` can be confusing.

Reply:
Done.  _(The examples were originally from a tool I wrote that generated AsyncAPI from Cloud Pub/Sub topology, wanted to make it clear where these values came from.)_